subworkflows/drep_large_catalogue_swf.nf

```diff
@@ -0,0 +1,128 @@
+/*
+ * De-replicate large catalogues > 100k genomes
+ */
+
+include { DREP_CHUNKED } from '../modules/drep_chunked'
+include { DREP_CHUNKED as DREP_RERUN } from '../modules/drep_chunked'
+include { COMBINE_CHUNKED_DREP } from '../modules/combine_chunked_drep'
+include { SPLIT_DREP } from '../modules/split_drep'
+include { CLASSIFY_CLUSTERS } from '../modules/classify_clusters'
+include { MASH_COMPARE } from '../modules/mash_compare'
+
+process COLLECT_DREP_RESULTS {
+
+    publishDir(
+        path: "${params.outdir}/additional_data/intermediate_files",
+        mode: "copy",
+        failOnError: true
+    )
+
+    stageInMode "copy"
+
+    input:
+    path drep_tables_tarballs
+    path cdb_csv
+    path sdb_csv
+    path mdb_csv
+
+    output:
+    path "drep_data_tables.tar.gz"
+
+    script:
+    """
+    tar -czf drep_data_tables.tar.gz \
+        ${cdb_csv} \
+        ${sdb_csv} \
+        ${mdb_csv} \
+        ${drep_tables_tarballs.join(' ')}
+    """
+
+}
+
+workflow DREP_LARGE_SWF {
+    take:
+        genomes_directory
+        checkm_csv
+        extra_weight_table
+
+    main:
+
+        // split genomes from genomes_directory into chunks 25000 files each (probably doesnt work)
+        genomes_chunked = genomes_directory
+            .map( { dir -> files("${dir}/*.fa") } )
+            .flatten()
+            .buffer( size: params.xlarge_chunk_size, remainder: true )
+
+        // dRep each chunk
+        DREP_CHUNKED(
+            genomes_chunked,
+            checkm_csv.first(),
+            extra_weight_table.first(),
+            false // not-merged - used for the output tarball name
+        )
+
+        dereplicated_genomes = DREP_CHUNKED.out.dereplicated_genomes.collect()
+
+        // re-run dDrep on species representatives
+        DREP_RERUN(
```

Comment:
Probably makes sense to use better name for DREP_RERUN. That one looks like something fail and we will re-run dRep.

Reply:
Yes, good suggestion. Maybe DREP_SPECIES_REPS?

Reply:
yes, sounds better :)